File: backend/src/main/java/com/go/ski/team/core/model/SkiResort.java
package com.go.ski.team.core.model;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import jakarta.persistence.*;
import lombok.*;

import static lombok.AccessLevel.PROTECTED;

@Entity
@Getter
@NoArgsConstructor(access = PROTECTED)
@ToString
@JsonIgnoreProperties({"hibernateLazyInitializer", "handler"})
public class SkiResort {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer resortId;

    @Column(nullable = false)
    private String resortName;

    @Column(nullable = false)
    private String resortLocation;

    @Column(nullable = false)
    private Double latitude;

    @Column(nullable = false)
    private Double longitude;

    @Builder
    public SkiResort(Integer resortId, String resortName, String resortLocation,
                     Double latitude, Double longitude){
        this.resortId = resortId;
        this.resortName = resortName;
        this.resortLocation = resortLocation;
        this.latitude = latitude;
        this.longitude = longitude;
    }

}

File: goski_student/lib/data/model/ski_resort.dart
class SkiResort {
  final int resortId;
  final String resortName;
  final String resortLocation;
  final double latitude;
  final double longitude;
  final List<int> lessonTimeList;

  SkiResort({
    required this.resortId,
    required this.resortName,
    required this.resortLocation,
    required this.latitude,
    required this.longitude,
    required this.lessonTimeList,
  });

  factory SkiResort.fromJson(Map<String, dynamic> json) {
    var list = json['lessonTime'] as List;
    List<int> lessonTimeList = list.map((i) => i as int).toList();

    return SkiResort(
      resortId: json['resortId'] as int,
      resortName: json['resortName'] as String,
      resortLocation: json['resortLocation'] as String,
      latitude: json['latitude'] as double,
      longitude: json['longitude'] as double,
      lessonTimeList: lessonTimeList,
    );
  }
}
